### ISSUE TITLE
Flake outputs are misleading.

### ISSUE THREAD

**Issue by shymega** [pimalaya/neverest#3]
The Flake is misleading. Packages should be `system`-specific, and not expose attrs of more `system`s.

Below is a screenshot. I would propose `arm64-linux` and `x86_64-windows` to be moved up a level.

![notright](https://github.com/user-attachments/assets/55eb0b48-21e2-4b15-a22a-bcaa84119969)

**Comment by shymega** [comment]
NB: I'm happy to make a PR. The issue is reproducible with Himalaya as well.

**Comment by soywod** [comment]
The structure is not well documented yet, but the first level correspond to host platforms, the second the cross platform. From ` x86_64-linux`, you can cross compile for `x86_64-linux`, `arm64-linux` and also `x86_64-windows`. I like the structure, it took me a while to set it up correctly. It just needs to be better documented and fixed (somehow does not work properly since last upgrade).

**Comment by shymega** [comment]
Ah... I see. Okay. That makes sense. Maybe, just to make it clear for people, you could add - either a prefix or a suffix - `-cross`.

That way it's obvious it's for cross-compiling. What are your thoughts on that?

**Comment by soywod** [comment]
It allows this kind of things in [CI](https://github.com/pimalaya/neverest/blob/fc5a999205f314e396814dc64cd321881106f8ca/.github/workflows/release.yml#L58), which I find great and I would like to keep it. It makes builds more predictable and reproducible.

**Comment by shymega** [comment]
Agreed. Wouldn't a suffix/prefix be relatively easy to adjust in CI as well, though? I mean, it wouldn't be 'typical' user-facing, only for those like me who inspect the Flake, and even then, with documentation in the flake, it would be explained.

**Comment by soywod** [comment]
> Maybe, just to make it clear for people, you could add - either a prefix or a suffix - `-cross`.

Yes indeed, I like the proposition!

**Comment by shymega** [comment]
Would you like me to make a PR?

**Comment by soywod** [comment]
As you wish. It should not take long for me to fix it, I already put my nose in recently. Thank you for proposing, feel free to submit other PRs if you want!

**Comment by shymega** [comment]
I've been thinking about this some more.

I may be wrong, but I believe nix/nixpkgs does support `system` attributes,
such as: `x86_64-windows`, and `aarch64-darwin`.

We could refactor the Flake here (and on the other projects) to export the
nested attributes up a level. It should work with Nix, but you may want to
check on the forums.

I think this would be simpler than the current approach and/or adding a
`-cross` suffix.

I suppose the ultimate goal is to upstream to Nixpkgs, right?

**Comment by soywod** [comment]
Sorry, I realized I've never replied to you.

> I think this would be simpler than the current approach and/or adding a `-cross` suffix.

Should be definitely enough, would you like to propose a PR?
 
> I suppose the ultimate goal is to upstream to Nixpkgs, right?

It's already available on Nixpkgs, but with another derivation. The best would be to use the same between both, do you think it's doable?

**Comment by soywod** [comment]
I think we can close the issue. The actual structure works well and is used across all Pimalaya tools. There is `cross-*` variants, and default (that does not cross):

```
git+file:///home/soywod/code/pimalaya/neverest?ref=refs/heads/master&rev=ae8d4aa3c4c560443ddbb0335d9d3bff29d4b0d0
├───devShells
│   ├───aarch64-darwin
│   │   └───default omitted (use '--all-systems' to show)
│   ├───aarch64-linux
│   │   └───default omitted (use '--all-systems' to show)
│   ├───x86_64-darwin
│   │   └───default omitted (use '--all-systems' to show)
│   └───x86_64-linux
│       └───default: development environment 'nix-shell'
└───packages
    ├───aarch64-darwin
    │   ├───cross-aarch64-darwin omitted (use '--all-systems' to show)
    │   └───default omitted (use '--all-systems' to show)
    ├───aarch64-linux
    │   ├───cross-aarch64-linux omitted (use '--all-systems' to show)
    │   └───default omitted (use '--all-systems' to show)
    ├───x86_64-darwin
    │   ├───cross-x86_64-darwin omitted (use '--all-systems' to show)
    │   └───default omitted (use '--all-systems' to show)
    └───x86_64-linux
        ├───cross-aarch64-linux: package 'neverest-static-aarch64-unknown-linux-musl-1.0.0-rc'
        ├───cross-armv6l-linux: package 'neverest-static-armv6l-unknown-linux-musleabihf-1.0.0-rc'
        ├───cross-armv7l-linux: package 'neverest-static-armv7l-unknown-linux-musleabihf-1.0.0-rc'
        ├───cross-i686-linux: package 'neverest-static-i686-unknown-linux-musl-1.0.0-rc'
        ├───cross-x86_64-linux: package 'neverest-static-x86_64-unknown-linux-musl-1.0.0-rc'
        ├───cross-x86_64-windows: package 'neverest-static-x86_64-w64-mingw32-1.0.0-rc'
        └───default: package 'neverest-1.0.0-rc'
```
